lib/faraday/options.rb

```diff
@@ -174,6 +174,7 @@ def self.memoized(key, &block)
 
       memoized_attributes[key.to_sym] = block
       class_eval <<-RUBY, __FILE__, __LINE__ + 1
+        remove_method(key) if method_defined?(key, false)
```

Comment:
🗒️ The `memoized` helper can be used to memoize existing methods, or it can be used to create entirely new methods. So it is necessary to check whether the method exists before attempting to remove it.

The specs have an example of `memoized` being used to synthesize a previously non-existent method `a`: https://github.com/lostisland/faraday/blob/d40f3dbb997bc12d8f89978f68eee3262b791fba/spec/faraday/options/options_spec.rb#L260